app/scripts/modules/core/navigation/urlBuilder.service.ts

```diff
@@ -7,18 +7,18 @@ export interface IUrlBuilderInput {
   account: string;
   application: string;
   cluster: string;
-  detail: string;
-  instanceId: string;
-  loadBalancer: string;
-  name: string;
-  project: string;
-  provider: string;
-  region: string;
-  serverGroup: string;
-  stack: string;
-  taskId: string;
+  detail?: string;
+  instanceId?: string;
+  loadBalancer?: string;
+  name?: string;
+  project?: string;
+  provider?: string;
+  region?: string;
+  serverGroup?: string;
+  stack?: string;
+  taskId?: string;
   type: string;
-  vpcId: string;
+  vpcId?: string;
```

Comment:
I'm questioning the validity of this interface. Seems like everything should be optional (when are you sending in a `taskId` and an `account`, for example), or we should refactor this into distinct interfaces to declare the expected parameters for expected returned URLs.

Not worth doing in this PR (this is fine for now), just something to consider, maybe add a TODO.